lib/pry/core_extensions.rb

```diff
@@ -86,7 +86,9 @@ def __binding__
       return class_eval("binding")
     end
 
-    unless respond_to?(:__pry__)
+    begin
+      __pry__
+    rescue
```

Comment:
Can this be `NoMethodError`?

Reply:
If an object claims that it has its own `__pry__` method but actually doesn't have it or has a buggy one as a result of some reckless `method_missing` magic, then invoking that method could either raise a `NoMethodError` or an `ArgumentError`. No idea about other errors.
Maybe, we should try to always define our `__pry__` method but this would probably lead to performance issues.